src/app/[locale]/specs/sync/page.mdx

```diff
@@ -0,0 +1,171 @@
+- services should ignore commit events with a `rev` lower or equal to the most recent processed `rev` for that DID, and should reject commit events with a `rev` corresponding to a future timestamp (beyond a clock drift window of a few minutes)
+- services should check the `since` value in commit events, and if it does not match the previous seen `rev` for that DID, mark the repo as out-of-sync (similar to a `tooBig` commit event)
+- data limits on records specifically should be verified. Events containing corrupt or entirely invalid records may be rejected. for example, a record not being CBOR at all, or exceeding normal data size limits.
+- more subtle data validation of records may be enforced, or may be ignored, depending on the service. For example, unsupported CID hash types embedded in records should probably be ignored by Relays (even if they violate the atproto data model), but may result in the record or commit event being rejected by an AppView
+- mirroring services, which retain a full copy of repository data, should verify that commit diffs leave the MST tree in a complete and valid state (eg, no missing records, no invalid MST nodes, commit CID would be reproducible if the MST structure was re-generated from scratch)
+- Relays (specifically) should not validate records against Lexicons
+
+## Reliable Synchronization
+
+This section describes some details on how to reliably subscribe to the firehose and maintain an existing synchronized mirror of the network.
+
+Services should generally maintain a few pieces of state for all accounts they are tracking data from:
+
+- track the most recent commit `rev` they have successfully processed
+- keep cached identity data, and use cache expiration to ensure periodic re-validation of that data
+- track account status
+
+Identity caches should be purged any time an `#identity` event is received. Additionally, identity resolution should be refreshed if a commit signature fails to verify, in case the signing key was updated but the identity cache has not been updated yet.
+
+When `tooBig` events are emitted on the firehose, downstream services will need to fetch the diff out-of-band. This usually means an API request to the `com.atproto.sync.getRepo` endpoint on the current PDS host for the account, with the `since` field included. The `since` value should be the most recently processed `rev` value for the account, which may or may not match the `since` field in the commit event message.
+
+If a `#commit` is received with a `since` that does not match the most recently processed `rev` for the account, and is “later” (higher value) than the most recent commit `rev` the service has processed for that account, the service may need to do the same kind of out-of-band fetch as for a `tooBig` event.
```

Comment:
```suggestion
If a `#commit` is received with a `since` that does not match the most recently processed `rev` for the account, and is “later” (higher value) than the most recent commit `rev` the service has processed for that account, there are likely commits missing or received out of order, which can usually be remedied by the out-of-band fetch prescribed above for each `tooBig` event received.
```